extras/traccc_throughput_mt_plotter.py

```diff
@@ -0,0 +1,441 @@
+                        continue
+                    # Add the result to the plot.
+                    throughput = getThroughput(row)
+                    profile.Fill(float(row['threads']), throughput)
+                    if throughput > max_throughput:
+                        max_throughput = throughput
+                        pass
+                    pass
+                pass
+
+            # Save the max throughput.
+            result.append([sample, input[1], max_throughput])
+
+            # Draw the profile on the canvas.
+            profile.Draw(drawOptions)
+            drawOptions = 'SAME %s' % drawOptions
+            pass
+
+        # Draw the legend.
+        legend.Draw()
+
+        # Save the page with all the profiles.
+        if firstPage:
+            canvas.Print('%s(' % output, 'pdf')
+            firstPage = False
+        else:
+            canvas.Print(output, 'pdf')
+            pass
+        pass
+
+    # Return the max throughputs for each sample and platform.
+    return result
+
+def drawMaxThroughputPlot(inputs, max_throughputs, samples, canvas, output,
+                          graphicsOptions):
+    '''Draw a single plot with the maximum throughputs on each platform
+
+    Argument(s):
+       inputs          -- List of CSV file name, platform name pairs
+       max_throughputs -- The throughputs collected in drawPerMuSamplePlots()
+       samples         -- Names of the (mu) samples to make plots for
+       canvas          -- The TCanvas to draw on
+       output          -- Output PDF file name
+       graphicsOptions -- The graphics options to use for the profile
+    '''
+
+    # Collect the platform names, and the minimum-maximum values.
+    platforms = []
+    for thr in max_throughputs:
+        if not thr[1] in platforms:
+            platforms.append(thr[1])
+            pass
+        pass
+
+    # Get the absolute minimum-maximum throughput values.
+    min_max_throughput = getMinMaxThroghput(inputs)
+
+    # The legend for all of the platforms used.
+    legend = ROOT.TLegend(0.5, 0.6, 0.9, 0.9)
+
+    # List of temporary profile objects. These need to stay alive until the
+    # canvas is printed.
+    profiles = []
+
+    # Helper counter.
+    counter = 0
+
+    # Helper variable for the draw option(s).
+    drawOptions = 'E1 P L'
+
+    # Display the maximum throughputs in logarithmic scale.
+    canvas.SetLogy()
+
+    # Create one profile per platform.
+    for platform in platforms:
+
+        # Create the profile.
+        profile = configureProfile(ROOT.TProfile(
+            'max_throughput_profile_%i' % counter,
+            'Maximum Throughput;;Events/Second',
+            len(samples), 0.5, len(samples) + 0.5), platform, graphicsOptions)
+        for i in range(len(samples)):
+            profile.GetXaxis().SetBinLabel(i + 1, samples[i])
+            pass
+        profile.GetYaxis().SetRangeUser(40., min_max_throughput[1] * 5.)
```

Comment:
If I understand this correctly, wouldn't defining the minimum like this be a better approach?
```suggestion
        profile.GetYaxis().SetRangeUser(min_max_throughput[0]*0.8, min_max_throughput[1] * 5.)
```

Reply:
Yes, I absolutely just hardcoded a value as I wanted to get something reasonable-looking.

I think your proposal of "some reasonable" multiplier is a good one. I'm afraid of making the code "too smart", but this is not that just yet. :stuck_out_tongue:

Reply:
:+1: